packages/darklang/cli/integrationTests/test-framework.dark

```diff
@@ -0,0 +1,283 @@
+                          | Some patternPart ->
+                            let patternEnd = Stdlib.String.split patternPart "'"
+                            let pattern = Stdlib.List.getAt patternEnd 0L
+                            match pattern with
+                            | Some p -> $"EOF: CLI process ended unexpectedly while waiting for '{p}'"
+                            | None -> "EOF: CLI process ended unexpectedly"
+                          | None -> "EOF: CLI process ended unexpectedly"
+                        else
+                          "EOF: CLI process ended unexpectedly"
+                      else
+                        // If it doesn't match our patterns, show the full error output
+                        if Stdlib.String.isEmpty errorOutput then
+                          "Interactive test failed with no error output"
+                        else
+                          $"Interactive test failed: {errorOutput}"
+
+                    TestUtils.printlnColored CliColors.red $"✗ FAIL - {interactiveTest.name}"
+                    TestUtils.printlnColored CliColors.red specificError
+                    TestResult.Failure $"Interactive test {interactiveTest.name} failed - {specificError}"
+
+
+        let runCliTest (config: CliTest): TestResult =
+          if config.isInteractive then
+            match config.interactiveTest with
+            | Some interactiveTest -> InteractiveTestRunner.runInteractiveTest interactiveTest
+            | None -> TestResult.Failure "Interactive test configuration missing"
+          else
+            let expectedFile = TestConfig.getExpectedFilePath config.expectedOutputFile
+            let actualFile = TestConfig.getActualFilePath config.name
+            let command = TestConfig.buildCliCommand config.arguments
+
+            match createDirectories() with
+            | Error e -> TestResult.Failure $"Failed to create directories: {e}"
+            | Ok _ ->
+              match runCliCommand command actualFile with
+              | Error e ->
+                TestUtils.printlnColored CliColors.red "✗ CLI command failed"
+                Builtin.printLine "CLI output:"
+                match TestUtils.getFileContent actualFile with
+                | Ok content when Stdlib.Bool.not (Stdlib.String.isEmpty content) ->
+                  Builtin.printLine content
+                | _ ->
+                  Builtin.printLine "No output captured"
+                TestResult.Failure $"CLI command failed: {e}"
+              | Ok _ ->
+                if Stdlib.Bool.not (isFileEmpty actualFile) then
+                  TestUtils.printlnColored CliColors.red "✗ Output file is empty."
+                  TestResult.Failure "Output file is empty"
+                else
+                  if Stdlib.Bool.not (TestUtils.fileExists expectedFile) then
+                    TestUtils.printlnColored CliColors.yellow $"⚠ Expected output file not found: {expectedFile}"
```

Comment:
might be good to add a `printlnWarning` shortcut, or similar? just to be a bit less verbose